common/level.go

```diff
@@ -57,17 +54,28 @@ type layer struct {
 
 func createTileset(lvl *Level, sheets []*tilesheet) []*tile {
 	tileset := make([]*tile, 0)
-	tw := lvl.TileWidth
-	th := lvl.TileHeight
+	tw := float32(lvl.TileWidth)
+	th := float32(lvl.TileHeight)
 
 	for _, sheet := range sheets {
-		setWidth := int(sheet.Image.Width()) / tw
-		setHeight := int(sheet.Image.Height()) / th
-		totalTiles := setWidth * setHeight
+		setWidth := sheet.Image.Width / tw
+		setHeight := sheet.Image.Height / th
+		totalTiles := int(setWidth * setHeight)
 
 		for i := 0; i < totalTiles; i++ {
 			t := &tile{}
-			t.Image = regionFromSheet(sheet.Image, tw, th, i)
+			setWidth := sheet.Image.Width / tw
+			x := float32(i%int(setWidth)) * tw
+			y := float32(i/int(setWidth)) * th
+
+			invTexWidth := 1.0 / float32(sheet.Image.Width)
+			invTexHeight := 1.0 / float32(sheet.Image.Height)
+
+			u := float32(x) * invTexWidth
+			v := float32(y) * invTexHeight
+			u2 := float32(x+tw) * invTexWidth
+			v2 := float32(y+th) * invTexHeight
+			t.Image = &Texture{id: sheet.Image.Texture, width: tw, height: th, viewport: engo.AABB{engo.Point{u, v}, engo.Point{u2, v2}}}
```

Comment:
You're not setting the `width` and `height` for `t`, nor the `engo.Point`. 

Are you using the anonymous `engo.Point` defined in line 39?